src/redux/channelActions.js

```diff
@@ -0,0 +1,28 @@
+import { API_BASE_URL } from "../backend-request/index";
+
+export const channel = {
+  SET_IS_CHANNELS_LOADING: "SET_IS_CHANNELS_LOADING",
+  SET_CHANNELS_LIST: "SET_CHANNELS_LIST"
+};
+
+export const setIsChannelsLoading = (isLoading = false) => ({
+  type: channel.SET_IS_CHANNELS_LOADING,
+  data: isLoading
+});
+export const setChannelList = (channels = []) => ({
+  type: channel.SET_CHANNELS_LIST,
+  data: channels
+});
+
+// fetch channels thunk 💡
+export const fetchChannelsThunk = () => dispatch => {
+  return fetch(API_BASE_URL + "/channels")
+    .then(res => res.json())
+    .then(data => {
+      dispatch(setIsChannelsLoading(false));
+      dispatch(setChannelList(data));
```

Comment:
I'd suggest collapsing these two actions, so that the reducer for `setChannelList` also sets the loading state to false. 

Also, this thunk should probably `dispatch(setIsChannelsLoading(true)` at the top, before the `fetch`.

Reply:
That's some nice refactoring.